Dockerfile

```diff
@@ -0,0 +1,325 @@
+#  _  ___                 _ ____
+# | |/ (_)_ __ ___   __ _(_)___ \
+# | ' /| | '_ ` _ \ / _` | | __) |
+# | . \| | | | | | | (_| | |/ __/
+# |_|\_\_|_| |_| |_|\__,_|_|_____|
+#
+
+# Source base [fpm/apache]
+ARG BASE="fpm"
+ARG PHP_VER="8.1"
+ARG COMPOSER_VER="latest"
+# Branch name
+ARG KIMAI="main"
+# Timezone for images
+ARG TIMEZONE="Europe/Berlin"
+
+###########################
+# Shared tools
+###########################
+
+# full kimai source
+FROM alpine:latest AS git-dev
+# pass-through Arguments in every stage. See: https://benkyriakou.com/posts/docker-args-empty
+ARG KIMAI
+ARG TIMEZONE
+# I need to do this check somewhere, we discard all but the checkout so doing here doesn't hurt
+ADD .docker/test-kimai-version.sh /test-kimai-version.sh
+RUN /test-kimai-version.sh
+RUN apk add --no-cache git && \
+    git clone --depth 1 --branch ${KIMAI} https://github.com/kimai/kimai.git /opt/kimai
+
+# production kimai source
+FROM git-dev AS git-prod
+
+WORKDIR /opt/kimai
+RUN rm -r tests
+
+# composer base image
+FROM composer:${COMPOSER_VER} AS composer
```

Comment:
Idea for the future: https://blog.codito.dev/2022/11/composer-binary-only-docker-images/
Makes builds faster and "cheaper, considering that we request a download of 3MB instead of 180MB